### PR TITLE
TECHX-577 create a shared docker-publish action

### PR DESCRIPTION
**Pull request by tommymorgan**

### CODE REVIEW

File: deploy-to-env/action.yaml [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+      repository: icanbwell/cie.gha-deploy
+      token: ${{ inputs.auth-token }}
+      path: .github/actions/cie.gha-deploy
+
+  - name: Deploy to Services
+    uses: ./.github/actions/cie.gha-deploy
+    with:
+      token: ${{ inputs.auth-token }}
+      service-name: ${{ inputs.service-name }}
+      env: 'services'
```
Comment (jaysonboyer) [inline deploy-to-env/action.yaml:45]: needs a parameter

File: docker-publish/action.yaml [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+  tag:
+    description: The tag to use
+    required: true
+    type: string
+  
+runs:
+  using: "composite"
+  # See also https://docs.docker.com/docker-hub/builds/
+  steps:
+    - name: Setup npmrc
```
Comment (jaysonboyer) [inline docker-publish/action.yaml:20]: maybe break this out into it's own action. if not I can do it later

File: deploy-to-env/action.yaml (hunk 1 above)
Comment (tommymorgan) [reply deploy-to-env/action.yaml]: fixed

**Review by jaysonboyer — APPROVED** [review]

File: docker-publish/action.yaml (hunk 2 above)
Comment (tommymorgan) [reply docker-publish/action.yaml:20]: I'll make this a stretch goal for tomorrow in a separate PR.

**Pull request merged**